nomad/search_endpoint.go

```diff
@@ -93,6 +94,252 @@ func (s *Search) getMatches(iter memdb.ResultIterator, prefix string) ([]string,
 	return matches, iter.Next() != nil
 }
 
+func (s *Search) getFuzzyMatches(iter memdb.ResultIterator, text string) (map[structs.Context][]structs.FuzzyMatch, map[structs.Context]bool) {
+	limitQuery := s.srv.config.SearchConfig.LimitQuery
+	limitResults := s.srv.config.SearchConfig.LimitResults
+
+	unsorted := make(map[structs.Context][]fuzzyMatch)
+	truncations := make(map[structs.Context]bool)
+
+	accumulateSet := func(limited bool, set map[structs.Context][]fuzzyMatch) {
+		for ctx, matches := range set {
+			for _, match := range matches {
+				if len(unsorted[ctx]) < limitResults {
```

Comment:
This is going to sound nitpicky, but the results limit config talks about a maximum number of results, whereas this is getting us the maximum number of results per context. Should we update this so that we're tracking a count of how many results overall, or just change the docs? (Either seems reasonable.)

Reply:
good catch; clarified in docs